pydis_site/apps/api/serializers.py

```diff
@@ -97,6 +97,16 @@ class Meta:
         fields = ('package', 'base_url', 'inventory_url')
 
 
+class AllowListSerializer(ModelSerializer):
+    """A class providing (de-)serialization of `AllowList` instances."""
+
+    class Meta:
+        """Metadata defined for the Django REST Framework."""
+
+        model = AllowList
+        fields = ('created_at', 'updated_at', 'type', 'allowed', 'content')
```

Comment:
I'm rusty with DRF. Will it allow `created_at` and `updated_at` to be set/changed even though they're auto fields? If yes, we should prevent that. Also, why are they listed here, but not the `id`?

Reply:
`id` is added in a later commit.

I've tried to POST requests with `updated_at` set to a datetime ten years in the past, but it seems to just ignore it and set it to what it should be anyway, so I don't think it's possible to set them. I think it's overridden by the auto field logics, or that the auto stuff happens later in the lifecycle.

Reply:
I think that this has less to do with DRF and more to do with Django: According to the documentation, `auto_now` and `auto_now_add` fields just ignore any value that you try to pass to it. This is what `auto_now_add` says:

> Note that the current date is always used; it’s not just a default value that you can override. So even if you set a value for this field when creating the object, it will be ignored. 

The comment for `auto_add` is similar, but does not include the `creating the object` line: It always just sets the current datetime when updating the model and will ignore any custom value you try to pass to it.